.github/workflows/opendata-ci.yml

```diff
@@ -0,0 +1,42 @@
+name: Open Data CI
+on:
+  pull_request:
+    branches:
+      - dev
+    paths:
+      - 'src/open-data/**/*.csv'
+      - 'src/open-data/**/*.json'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env: 
+      CI_COMMIT_MESSAGE: Update open data
+      CI_COMMIT_AUTHOR: Open Data CI
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.ref }}
+        token: ${{ secrets.WORKFLOW_GIT_ACCESS_TOKEN }}
+    - name: Set environment variable "commit-message"
+      run: echo "commit-message=$(git log -1 --pretty=format:'%s')" >> $GITHUB_ENV
+    - name: Set environment variable "commit-author"
+      run: echo "commit-author=$(git log -1 --pretty=format:'%an')" >> $GITHUB_ENV
+    - name: Set environment variable "is-auto-commit"
+      if: env.commit-message == env.CI_COMMIT_MESSAGE && env.commit-author == env.CI_COMMIT_AUTHOR
+      run: echo "is-auto-commit=true" >> $GITHUB_ENV
+    - name: Build Open Data
+      if: env.is-auto-commit == false
+      shell: pwsh
+      run: |
+        src/scripts/Build-OpenData.ps1 -PowerShell
+        if (git status --porcelain)
+        {
+          git config --global user.name "${{ env.CI_COMMIT_AUTHOR }}"
+          git config --global user.email "finopstoolkit@users.noreply.github.com"
+          git commit -a -m "${{ env.CI_COMMIT_MESSAGE }}"
+          git push
+        }
+    - name: Test Open Data
+      shell: pwsh
+      run: |
+        src/scripts/Build-OpenData.ps1 -Test
```

Comment:
Do we need to do anything with the results? Does the action fail if any tests fail? 

Is there any way to write a comment to the PR to convey what failed? I know we can look at the logs, but that's a bit tedious 😜 We can add a TODO comment for this, if it makes sense but you want to cover it later. It's not a blocker in any way.

Reply:
The action will fail if the tests fail yes. I can look into adding a comment on failures for sure.

Reply:
I added a task for auto comments in the issue for now